confignode/src/main/java/org/apache/iotdb/confignode/persistence/sync/ClusterSyncInfo.java

```diff
@@ -0,0 +1,104 @@
+import org.apache.iotdb.commons.snapshot.SnapshotProcessor;
+import org.apache.iotdb.commons.sync.metadata.SyncMetadata;
+import org.apache.iotdb.confignode.consensus.request.write.sync.CreatePipeSinkPlan;
+import org.apache.iotdb.confignode.consensus.request.write.sync.DropPipeSinkPlan;
+import org.apache.iotdb.confignode.consensus.request.write.sync.GetPipeSinkPlan;
+import org.apache.iotdb.confignode.consensus.response.PipeSinkResp;
+import org.apache.iotdb.db.utils.sync.SyncPipeUtil;
+import org.apache.iotdb.rpc.RpcUtils;
+import org.apache.iotdb.rpc.TSStatusCode;
+
+import org.apache.commons.lang3.StringUtils;
+import org.apache.thrift.TException;
+import org.slf4j.Logger;
+import org.slf4j.LoggerFactory;
+
+import java.io.File;
+import java.io.IOException;
+import java.util.Collections;
+
+public class ClusterSyncInfo implements SnapshotProcessor {
+
+  protected static final Logger LOGGER = LoggerFactory.getLogger(ClusterSyncInfo.class);
+
+  private final SyncMetadata syncMetadata;
+
+  public ClusterSyncInfo() {
+    syncMetadata = new SyncMetadata();
+  }
+  // ======================================================
+  // region Implement of PipeSink
+  // ======================================================
+
+  public void checkAddPipeSink(String pipeSinkName) throws PipeSinkException {
+    syncMetadata.checkAddPipeSink(pipeSinkName);
+  }
+
+  public TSStatus addPipeSink(CreatePipeSinkPlan plan) {
+    TSStatus status = new TSStatus();
+    try {
+      syncMetadata.addPipeSink(SyncPipeUtil.parsePipeInfoAsPipe(plan.getPipeSinkInfo()));
+      status.setCode(TSStatusCode.SUCCESS_STATUS.getStatusCode());
+    } catch (PipeSinkException e) {
+      status.setCode(TSStatusCode.PIPESINK_ERROR.getStatusCode());
+      LOGGER.error(e.getMessage());
+    }
+    return status;
+  }
+
+  public void checkDropPipeSink(String pipeSinkName) throws PipeSinkException {
+    syncMetadata.checkDropPipeSink(pipeSinkName);
+  }
+
+  public TSStatus dropPipeSink(DropPipeSinkPlan plan) {
+    syncMetadata.dropPipeSink(plan.getPipeSinkName());
+    return RpcUtils.getStatus(TSStatusCode.SUCCESS_STATUS);
+  }
+
+  public PipeSinkResp getPipeSink(GetPipeSinkPlan plan) {
+    PipeSinkResp resp = new PipeSinkResp();
+    if (StringUtils.isEmpty(plan.getPipeSinkName())) {
+      resp.setPipeSinkList(syncMetadata.getAllPipeSink());
+    } else {
+      resp.setPipeSinkList(
+          Collections.singletonList(syncMetadata.getPipeSink(plan.getPipeSinkName())));
+    }
+    resp.setStatus(RpcUtils.getStatus(TSStatusCode.SUCCESS_STATUS));
+    return resp;
+  }
+
+  @Override
+  public boolean processTakeSnapshot(File snapshotDir) throws TException, IOException {
+    // TODO: merge snapshot logic into SyncLogWritter and SyncLogReader
+    return true;
+  }
+
+  @Override
+  public void processLoadSnapshot(File snapshotDir) throws TException, IOException {
+    // TODO: merge snapshot logic into SyncLogWritter and SyncLogReader
+  }
```

Comment:
Should add ClusterSyncInfoTest later. Better leave a TODO here in case of forgetting.

Reply:
done